src/vs/workbench/contrib/terminal/browser/links/terminalLinkManager.ts

```diff
@@ -141,12 +143,20 @@ export class TerminalLinkManager extends DisposableStore {
 		return links[0];
 	}
 
-	async getLinks(): Promise<IDetectedLinks> {
+	async getLinks(extended?: boolean): Promise<IDetectedLinks> {
 		const wordResults: ILink[] = [];
 		const webResults: ILink[] = [];
 		const fileResults: ILink[] = [];
-
-		for (let i = this._xterm.buffer.active.length - 1; i >= this._xterm.buffer.active.viewportY; i--) {
+		let noMoreResults: boolean = false;
+		let topLine = !extended ? this._xterm.buffer.active.viewportY - Math.min(this._xterm.rows, 50) : this._lastTopLine! - (Math.min(this._xterm.rows, 50));
+		if (topLine < 0 || topLine - Math.min(this._xterm.rows, 50) < 0) {
+			noMoreResults = true;
+		}
+		if (topLine < 0) {
+			topLine = 0;
+		}
+		this._lastTopLine = topLine;
```

Comment:
This looks like fixed size paging, can we do something more aggressive here like pull in 1000 per page or more if performance seems fine?

Reply:
I worry that pulling in a non-standard number might be surprising/unexpected from an accessibility perspective. I think it makes sense to have fixed size paging here.

Reply:
will add a show all

Reply:
will make the page 1000